### PR TITLE
Migrate screens/profiler code to null safety

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes https://github.com/flutter/devtools/issues/3776

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart [hunk 1]
```diff
@@ -56,37 +54,40 @@ class CpuProfilerController
...
   CpuProfileStore get cpuProfileStore =>
       _cpuProfileStoreByIsolateId.putIfAbsent(
-        serviceManager.isolateManager.selectedIsolate.value?.id,
+        serviceManager.isolateManager.selectedIsolate.value?.id ?? '',
         () => CpuProfileStore(),
       );
 
   /// Store of cached CPU profiles for each isolate.
-  final _cpuProfileStoreByIsolateId = <String, CpuProfileStore>{};
+  final Map<String, CpuProfileStore> _cpuProfileStoreByIsolateId =
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart:67]: Any reason for adding a type on the LHS?

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart [hunk 2]
```diff
@@ -91,9 +89,9 @@ class _CpuProfilerState extends State<CpuProfiler>
         TickerProviderStateMixin,
         AutoDisposeMixin,
         SearchFieldMixin<CpuProfiler> {
-  TabController _tabController;
+  TabController? _tabController;
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:92]: Can we make this late and non-nullable?

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart [hunk 3]
```diff
@@ -317,7 +305,7 @@ class _CpuProfilerState extends State<CpuProfiler>
     );
     // TODO(kenz): make this order configurable.
     return [
-      if (widget.summaryView != null) widget.summaryView,
+      if (widget.summaryView != null) widget.summaryView!,
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:308]: We can remove the redundant null check by putting `widget.summaryView` into a local before this return.

File: packages/devtools_app/test/cpu_profile_model_test.dart [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 // @dart=2.9
 
+// ignore_for_file: avoid_redundant_argument_values
```
Comment (bkonyi) [inline packages/devtools_app/test/cpu_profile_model_test.dart:7]: What's this for?

Comment (kenzieschmoll) [reply packages/devtools_app/test/cpu_profile_model_test.dart:7]: This lint can be triggered for files that have not been migrated to null safety that are importing null safe files. Once this file is migrated, the ignore can be removed.

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:92]: done

File: packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profiler.dart:308]: done

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart:67]: removed

**Review by bkonyi — APPROVED** [review]

**Pull request merged**